### PR TITLE
Update Blazor Identity scaffolding guidance

### LINKED ISSUE #32351: The option for identity scaffold is missing for blazor web app
### Description

> The option for identity scaffold is missing for blazor web app

### Page URL

https://learn.microsoft.com/en-au/aspnet/core/security/authentication/scaffold-identity?view=aspnetcore-8.0&tabs=visual-studio

### Content source URL

https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/security/authentication/scaffold-identity.md

### Document ID

3e25547b-e2c9-6634-2c38-2a77207f17d4

### Article author

@Rick-Anderson

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #32351

Notes ...

* There were a couple of NITs on the RCL naming: It should be all caps ("Razor Class Library") only when referring directly to the project template. Otherwise, it appears with a proper adjective for "Razor" and lowercase "class library" ("Razor class library") ... and the acronym "RCL" after first mention is fine. This is per DR's instructions ... *I lost the argument on it* 🙈😆.
* For Blazor, the Identity components are all just dumped into the app. There's no further pick-and-choose scaffolding. Therefore, there's only a *Scaffold Identity into a Blazor project* section being added here.
* Prior Blazor coverage for <8.0 is fine in its present form.
* The lead-in remarks for 8.0+ must change for Blazor, so I created a pair of sections at the top of the article. It might be possible to combine them, but perhaps this is OK like this. See what you think.
* DR asked me to use the **Connected Services UI** for EF Core database migrations in the upcoming Blazor tutorial, not the CLI/PS commands. I did that here, but I only did it for the Blazor section. Would you like me to copy-paste that for the 8.0+ RP/MVC coverage?
* Because this article doesn't have VSC coverage, I didn't add that. If this article ever gets RP/MVC VSC coverage, I can supply the text for the 8.0+ BWA scenario. I'll be 👂 for a ping if the article ever adopts VSC coverage.

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/razor-pages/ui-class.md](https://github.com/dotnet/AspNetCore.Docs/blob/ad272588d0a8183745b56afd5c859f1fe7bc3856/aspnetcore/razor-pages/ui-class.md) | [Create reusable UI using the Razor class library project in ASP.NET Core](https://review.learn.microsoft.com/en-us/aspnet/core/razor-pages/ui-class?branch=pr-en-us-33134) |
| [aspnetcore/security/authentication/scaffold-identity.md](https://github.com/dotnet/AspNetCore.Docs/blob/ad272588d0a8183745b56afd5c859f1fe7bc3856/aspnetcore/security/authentication/scaffold-identity.md) | [Scaffold Identity in ASP.NET Core projects](https://review.learn.microsoft.com/en-us/aspnet/core/security/authentication/scaffold-identity?branch=pr-en-us-33134) |

### CODE REVIEW

File: aspnetcore/security/authentication/scaffold-identity.md [hunk 1]
```diff
@@ -16,7 +16,19 @@ By [Rick Anderson](https://twitter.com/RickAndMSFT)
...
+## Blazor Identity scaffolding
+
+ASP.NET Core Identity scaffolding adds [ASP.NET Core Identity](xref:security/authentication/identity) to Blazor Web Apps and Blazor Server apps. Identity Razor components added to the app can be customized to change their behavior.
+
+Although the scaffolder generates the necessary C# code to scaffold Identity into the app, you must update the project's database with an Entity Framework (EF) Core database migration to complete the process. This article explains the steps required to migrate a database.
+
+We recommend using a source control system that shows file differences and allows you to back out of changes. Inspect the changes after running the Identity scaffolder.
```
Comment (Rick-Anderson) [inline aspnetcore/security/authentication/scaffold-identity.md:25]: Just a thought

```suggestion
Inspect the changes after running the Identity scaffolder. We recommend using a GitHub or another source control system that shows file changes and provides a way to revert changes.
```

File: aspnetcore/security/authentication/scaffold-identity.md [hunk 2]
```diff
@@ -16,7 +16,19 @@ By [Rick Anderson](https://twitter.com/RickAndMSFT)
...
+
+Services are required when using [two-factor authentication (2FA)](xref:blazor/security/server/qrcodes-for-authenticator-apps), [account confirmation and password recovery](xref:blazor/security/server/account-confirmation-and-password-recovery), and other security features with Identity. Services or service stubs aren't generated when scaffolding Identity. Services to enable these features must be added manually.
+
+## Razor Pages and MVC Identity scaffolding
+
+ASP.NET Core provides [ASP.NET Core Identity](xref:security/authentication/identity) as a [Razor class library (RCL)](xref:razor-pages/ui-class). Applications that include Identity can apply the scaffolder to selectively add the source code contained in the Identity RCL. You might want to generate source code so you can modify the code and change the behavior. For example, you could instruct the scaffolder to generate the code used in registration. Generated code takes precedence over the same code in the Identity RCL. To gain full control of the UI and not use the default RCL, see the section [Create full Identity UI source](#full).
```
Comment (Rick-Anderson) [inline aspnetcore/security/authentication/scaffold-identity.md:31]: Your's is fine, feel free to ignore.

```suggestion
ASP.NET Core provides [ASP.NET Core Identity](xref:security/authentication/identity) as a [Razor class library (RCL)](xref:razor-pages/ui-class). Applications that include Identity can apply the scaffolder to selectively add the source code contained in the Identity RCL. You might want to generate source code so you can modify the code and change the behavior. For example, you could instruct the scaffolder to generate the code used in registration. Customized Identity code overrides the default implementation provided by the Identity Razor Class Library (RCL). To gain full control of the UI and not use the default RCL, see the section [Create full Identity UI source](#full).
```

Or something like Scaffoled Identity code supersedes the RCL defaults.

File: aspnetcore/security/authentication/scaffold-identity.md [hunk 3]
```diff
@@ -28,6 +40,124 @@ Services are required when using [Two Factor Authentication](xref:security/authe
...
+  * Select or add with the plus (**+**) button the database context class (**DbContext class**).
+  * Select the database provider (**Database provider**), which defaults to SQL Server.
+  * Select or add with the plus (**+**) button the user class (**User class**).
+  * Select the **Add** button.
+
+# [.NET CLI](#tab/net-cli)
+
+To add the required NuGet packages and tools, execute the following .NET CLI commands in a command shell opened to the project's root folder.
+
+Paste all of the following commands at the prompt (`>`) of the command shell. When you paste multiple commands, a warning appears stating that multiple commands will execute. Dismiss the warning and proceed with the paste operation.
```
Comment (Rick-Anderson) [inline aspnetcore/security/authentication/scaffold-identity.md:63]: Do you dismiss or ignore? IIRC, there is nothing to dismiss. When I read "Dismiss", I'm thinking I must take action other.

File: aspnetcore/security/authentication/scaffold-identity.md [hunk 4]
```diff
@@ -28,6 +40,124 @@ Services are required when using [Two Factor Authentication](xref:security/authe
...
+
+When you paste multiple commands, all of the commands execute except the last one. The last command doesn't execute until you press <kbd>Enter</kbd> on the keyboard.
```
Comment (Rick-Anderson) [inline aspnetcore/security/authentication/scaffold-identity.md:65]: I think execute is flagged by poly woke check?

```suggestion
When you paste multiple commands, all but the last one run. Press <kbd>Enter</kbd> to run the last command.
```

**Review by Rick-Anderson — APPROVED** [review]
> * DR asked me to use the **Connected Services UI** for EF Core database migrations in the upcoming Blazor tutorial, not the CLI/PS commands. I did that here, but I only did it for the Blazor section. Would you like me to copy-paste that for the 8.0+ RP/MVC coverage?

Please.

The only suggestion you need to review is execute. All the rest you can ignore if you like. As always, great work.

File: aspnetcore/security/authentication/scaffold-identity.md (hunk 3 above)
Comment (guardrex) [reply aspnetcore/security/authentication/scaffold-identity.md:63]: I can fix this ... the actual button text to get rid of the warning is **Paste anyway**, so I'll call that button out by name ...

```suggestion
Paste all of the following commands at the prompt (`>`) of the command shell. When you paste multiple commands, a warning appears stating that multiple commands will execute. Select the **Paste anyway** button.
```

**Comment by guardrex** [comment]
@Rick-Anderson ... I think all that I needed to do was update the `migrations.md` INCLUDE file with the Connected Services UI steps, using the migration name that you had there (`CreateIdentitySchema`).

I'm going to sleep 🛌💤 on this and make a final edit pass in the morning.

**Pull request merged**